packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -46,9 +59,35 @@ export function MultipleConnectionSidebar({}: MultipleConnectionSidebarProps) {
     []
   );
 
-  const onNewConnection = useCallback(() => {
-    // TODO: COMPASS-7710
-  }, []);
+  const onNewConnectionOpen = useCallback(
+    () => setIsConnectionFormOpen(true),
+    []
+  );
+  const onNewConnectionClose = useCallback(
+    () => setIsConnectionFormOpen(false),
+    []
+  );
+  const onNewConnectionToggle = useCallback(
+    (open: boolean) => setIsConnectionFormOpen(open),
+    []
+  );
+
+  const onNewConnectionConnect = useCallback(
+    (connectionInfo) => {
+      connect({
+        ...cloneDeep(connectionInfo),
+      }).then(() => setIsConnectionFormOpen(false));
```

Comment:
we had no error handling in https://github.com/mongodb-js/compass/blob/af95fc323d063384200ada17ff7f80ebdb8df890/packages/compass-connections/src/components/connections.tsx#L245, I assume this is because we'll have the connection progress bar @kmruiz

Reply:
Yep, we have error handling in the task I'm working on to show the error toast and a way to open the modal on edit mode.